syntax = "proto3";

package calculator;

service Calculator {
  rpc Add(Request) returns (Response) {}
  rpc Multiply(Request) returns (Response) {}
  rpc Subtract(Request) returns (Response) {}
  rpc Divide(Request) returns (Response) {}
}

message Request {
  int32 first_number = 1;
  int32 second_number = 2;
}

message Status {
  int32 code = 1;
  string message = 2;
}

message Response {
  int32 result = 1;
  Status status = 2;
}




